server/apple_notification_server.go

```diff
@@ -269,3 +262,23 @@ func (me *AppleNotificationServer) SendNotification(msg *PushNotification) PushR
 	}
 	return NewOkPushResponse()
 }
+
+func (me *AppleNotificationServer) SendNotificationWithRetry(notification *apns.Notification, retry int) (*apns.Response, error) {
+	start := time.Now()
+
+	ctx, cancel := context.WithTimeout(context.Background(), me.sendTimeout)
+	defer cancel()
+
+	res, err := me.AppleClient.PushWithContext(ctx, notification)
+	if me.metrics != nil {
+		me.metrics.observerNotificationResponse(PushNotifyApple, time.Since(start).Seconds())
+	}
+
+	if err != nil {
+		if nextIteration := retry + 1; nextIteration < MAX_RETRIES {
```

Comment:
Here we are not checking if we should retry the same way we do on Android

Reply:
On iOS the library works slightly different.

We may get a response from the library saying that it was not sent, and those would be the "known errors". Anything that comes in the `err` variable is something probably related to the network, so that is what we are retrying here.

Reply:
Hmmm I see, even when there is a mismatch with the device id or the device is no longer registered (for example the app has been unistalled)

Reply:
Yes, those scenarios are covered inside the response, and not in the error.